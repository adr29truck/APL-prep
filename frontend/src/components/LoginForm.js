// import {useRef} from 'react'

export const LoginForm = (props) => {
  const fields = props.fields.map(field => <input type={field.type} placeholder={field.placeholder || field.name} name={field.name} key={field.name}></input>)


  return (
    <form onSubmit={props.onSubmit}>    
      {fields}
      <button type="submit">Submit</button>
    </form>
  )
}

LoginForm.defaultProps = {
  fields: [
          {name: 'email', type: 'email'}, 
          {name: 'password', type: 'password'}
        ],
  onSubmit: (e) => {e.preventDefault(); console.log('A form was submitted')}
}