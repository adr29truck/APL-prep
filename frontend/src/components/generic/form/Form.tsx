/* eslint-disable react/require-default-props */
import { TextField, capitalize, InputLabel } from '@material-ui/core';
import React from 'react';

interface X {
  type: string;
  placeholder?: string;
  name: string;
  value: string | number;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

interface Props {
  fields: X[];
  onSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
}

const Form = ({ fields, onSubmit }: Props) => {
  const fieldsJsx = fields.map((field) => (
    <div key={`form_key${field.name}`}>
      <InputLabel htmlFor={`input_field_${field.name}`}>
        {capitalize(field.name)}
      </InputLabel>
      <TextField
        id={`input_field_${field.name}`}
        name={field.name}
        onChange={field.onChange}
        placeholder={field.placeholder || capitalize(field.name)}
        type={field.type}
        value={field.value}
        inputProps={{ 'aria-label': field.placeholder }}
      />
    </div>
  ));

  return (
    <form onSubmit={onSubmit}>
      {fieldsJsx}
      <button type="submit">Submit</button>
    </form>
  );
};

export default Form;
