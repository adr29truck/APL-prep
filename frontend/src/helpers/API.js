import axios from 'axios';

const baseUrl = () => {
  try {
    return `${window.location.protocol}//${window.location.hostname}:5000/`;
  } catch {
    return 'http://localhost:5000/';
  }
};

/**
 * Represents everythin API related
 */
export default class API {
  /**
   * Does get requests to the API
   * @param {string} route - string representation of API endpoint after baseUrl
   * @return {object} JSON response from API request
   */
  static get(route) {
    return axios.get(baseUrl() + route)
        .then((res) => {
          return res.data;
        }).catch((error) => {
          console.log(error);
        });
  }

  /**
   * Does post requests to the API
   * @param {string} route - string representation of API endpoint after baseUrl
   * @param {object} data - data to pass to the API
   * @return {object} JSON response from API request
   */
  static post(route, data) {
    return axios.post(baseUrl() + route, data)
        .then((res) => {
          return res.data;
        }).catch((error) => {
          console.log(error);
        });
  }
}
