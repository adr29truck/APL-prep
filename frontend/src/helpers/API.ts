/* eslint-disable no-console */
import axios, { AxiosResponse, AxiosError, AxiosRequestConfig } from 'axios';

const baseUrl = () => {
  try {
    return `${window.location.protocol}//${window.location.hostname}:5000/`;
  } catch {
    return 'http://localhost:5000/';
  }
};

const axiosConfig: AxiosRequestConfig = {
  headers: {
    'content-Type': 'application/json',
    Accept: '/',
    'Cache-Control': 'no-cache',
    // Authorization: localStorage.getItem('jwt'),
    // FIXME: IM BROKEN
    Authorization:
      'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpYXQiOjE2MTEyMzk0NjUsImV4cCI6MTYxMTMyNTg2NSwianRpIjoiYjJhNDY5YTItZGE1My00MGM1LWExNDYtMDg1ZTIxNzBkOTYwIiwiaWQiOjEsInJscyI6IiIsInJmX2V4cCI6MTYxMzgzMTQ2NX0.s702zGzynnTmyBgc0VzB6HJvccIRPMrH75pQ0YcR0fY',
  },
  // credentials: 'true',
};

/**
 * Represents everythin API related
 */
export default class API {
  /**
   * Does get requests to the API
   * @param {string} route - string representation of API endpoint after baseUrl
   * @return {object} JSON response from API request
   */
  static get(route: string) {
    return axios
      .get(baseUrl() + route, axiosConfig)
      .then((res: AxiosResponse) => {
        return res.data;
      })
      .catch((error: AxiosError) => {
        console.log(error);
      });
  }

  /**
   * Does post requests to the API
   * @param {string} route - string representation of API endpoint after baseUrl
   * @param {object} data - data to pass to the API
   * @return {object} JSON response from API request
   */
  static post(route: string, data: object) {
    return axios
      .post(baseUrl() + route, data, axiosConfig)
      .then((res: AxiosResponse) => {
        return res.data;
      })
      .catch((error: AxiosError) => {
        console.log(error);
      });
  }
}
