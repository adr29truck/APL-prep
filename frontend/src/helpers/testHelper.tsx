import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../store';

export const TestHelper = ({ children }: any) => {
  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
};
