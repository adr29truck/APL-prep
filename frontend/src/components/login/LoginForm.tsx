/* eslint-disable react/require-default-props */
import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import Form from '../generic/form';

const LoginForm = () => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  function onUsernameChange({ target }: React.ChangeEvent<HTMLInputElement>) {
    setUsername(target.value);
  }
  function onPasswordChange({ target }: React.ChangeEvent<HTMLInputElement>) {
    setPassword(target.value);
  }

  const dispatch = useDispatch();

  function onLoginFormSubmit(e: React.FormEvent<HTMLFormElement>): void {
    e.preventDefault();
    dispatch({ type: 'auth/signIn', payload: { username, password } });
    setUsername('');
    setPassword('');
  }

  return (
    <Form
      fields={[
        {
          name: 'username',
          type: 'text',
          value: username,
          onChange: onUsernameChange,
        },
        {
          name: 'password',
          type: 'password',
          value: password,
          onChange: onPasswordChange,
        },
      ]}
      onSubmit={onLoginFormSubmit}
    />
  );
};

export default LoginForm;
