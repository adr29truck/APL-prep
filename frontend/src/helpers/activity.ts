import API from './API';
import store from '../store';

export async function fetchActivities() {
  const temp = await API.get('activities');
  store.dispatch({ type: 'activity/setActivitiesState', payload: temp });
}
