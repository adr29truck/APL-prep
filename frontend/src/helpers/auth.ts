import store from '../store';
import API from './API';

/**
 * Extracts the content from a JWT token
 * @param {*} token - x
 * @return {Object} - data
 */
export function parseJwt(token: string) {
  const base64Url = token.split('.')[1];
  const base64 = base64Url.replace(/-/g, '+').replace(/_/g, '/');
  const jsonPayload = decodeURIComponent(
    atob(base64)
      .split('')
      .map((c) => {
        return `%${`00${c.charCodeAt(0).toString(16)}`.slice(-2)}`;
      })
      .join('')
  );
  return JSON.parse(jsonPayload);
}

export async function signIn(payload: object) {
  const temp = await API.post('api/login', payload);
  store.dispatch({ type: 'auth/setState', payload: temp });
}

export async function signOut() {
  await API.post('api/logout');
  store.dispatch({ type: 'auth/setState', payload: { id: 0 } });
}
