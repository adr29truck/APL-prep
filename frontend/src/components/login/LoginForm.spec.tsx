import React from 'react';
import { mount } from 'enzyme';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import '@testing-library/jest-dom/extend-expect';
import LoginForm from './LoginForm';
import Form from '../generic/form/Form';
import { TestHelper } from '../../helpers/testHelper';

describe('<Login />', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <TestHelper>
        <LoginForm />
      </TestHelper>
    );
    expect(wrapper.find(Form).html()).toMatchSnapshot();
  });
  it('handles form submission correctly', () => {
    render(
      <TestHelper>
        <LoginForm />
      </TestHelper>
    );
    userEvent.type(screen.getByPlaceholderText('Username'), 'admin');
    userEvent.type(screen.getByPlaceholderText('Password'), 'admin');
    expect(screen.getByRole('form')).toHaveFormValues({
      username: 'admin',
      password: 'admin',
    });

    userEvent.click(screen.getByText('Submit'));
    expect(screen.getByRole('form')).toHaveFormValues({
      username: '',
      password: '',
    });
  });
});
