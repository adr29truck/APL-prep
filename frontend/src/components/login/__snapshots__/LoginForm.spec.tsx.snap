// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `"<form name=\\"form\\"><div><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated\\" for=\\"input_field_username\\">Username</label><div class=\\"MuiFormControl-root MuiTextField-root\\"><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" id=\\"input_field_username\\" name=\\"username\\" placeholder=\\"Username\\" type=\\"text\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"\\"></div></div></div><div><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated\\" for=\\"input_field_password\\">Password</label><div class=\\"MuiFormControl-root MuiTextField-root\\"><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl\\"><input aria-invalid=\\"false\\" id=\\"input_field_password\\" name=\\"password\\" placeholder=\\"Password\\" type=\\"password\\" class=\\"MuiInputBase-input MuiInput-input\\" value=\\"\\"></div></div></div><button type=\\"submit\\">Submit</button></form>"`;
